src/util/returnUrlHelper.js

```diff
@@ -4,18 +4,32 @@ const generateReturnUrl = ({ queryParams, isMyAccount }) => {
   const {
     publisherConfig: { adyenConfiguration }
   } = store.getState();
+  let queryParamsString;
 
-  if (isMyAccount && adyenConfiguration?.myaccountReturnUrl)
-    return adyenConfiguration.myaccountReturnUrl;
+  if (isMyAccount && adyenConfiguration?.myaccountReturnUrl) {
+    if (queryParams) {
+      queryParamsString = Object.keys(queryParams).map(
+        key => `${key}=${queryParams[key]}`
+      );
+    }
+    return `${adyenConfiguration.myaccountReturnUrl}?${queryParamsString}`;
```

Comment:
what in case myaccountReturnUrl (or checkoutReturnUrl) will already have query params? Then we will end up with something like `https://client-page.com/checkout?somecustomparam=123?orderId=12345678`, while we should have  `https://client-page.com/checkout?somecustomparam=123&orderId=12345678`. We should detect there if we should pass '?' or '&'